net/dnscrypt-proxy/Portfile

```diff
@@ -1,34 +1,52 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 
 PortSystem          1.0
+PortGroup github    1.0
 
-name                dnscrypt-proxy
-version             1.9.5
-revision            1
-categories          net
+github.setup        jedisct1 dnscrypt-proxy 2.0.22
 license             ISC
-maintainers         {mps @Schamschula} openmaintainer
+revision            0
+categories          net
+maintainers         nomaintainer
 description         A protocol for securing communications between a client and a DNS \
                     resolver.
 long_description    ${description}
 homepage            https://dnscrypt.org/
-master_sites        https://download.dnscrypt.org/dnscrypt-proxy/
-
 platforms           darwin
-depends_lib         port:libsodium
 
-checksums           rmd160  836062bba81ba84ce33981696796bb163f559440\
-                    sha256  64021fabb7d5bab0baf681796d90ecd2095fb81381e6fb317a532039025a9399
+depends_build       port:go
+
+github.tarball_from tags
+
+checksums           rmd160  058bbfc0229ce8cece57f847805952e44180af23\
+                    sha256  4eb6dc3875a37f6c43cd1e3ba4d843a71121c5ef202b5eee7a770e018a491e4c\
+                    size    2193346
+
+use_configure       no
+
+worksrcdir          ${distname}/${name}
+build.cmd           go build
+build.args          -ldflags="-s -w"
+build.target
+build.env           GOPATH=${workpath}
```

Comment:
Actually there is also a golang portgroup. It will create these build commands for you.